# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libpref/Preferences.h
# Commit: 2adbab502a1e
# Full Hash: 2adbab502a1e690fccd61fb604be820e60d61723
# Author: Kristen Wright <kwright@mozilla.com>
# Date: 2020-09-09 09:39:57
# Regressor Bug: 1642727
# File Overlap Count: 1
# Description:
#   Bug 1642727 - Remove the VarCache machinery r=njn
#   
#   This commit removes Varcache entirely, putting an end to our quest to get rid of VarCache.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79538
# ==============================================================================

diff -r 1aeb7707569f -r 2adbab502a1e modules/libpref/Preferences.h
--- a/modules/libpref/Preferences.h	Tue Sep 08 15:05:06 2020 +0000
+++ b/modules/libpref/Preferences.h	Tue Sep 08 17:12:01 2020 +0000
@@ -393,84 +393,6 @@
                               PrefixMatch);
   }
 
-  // Adds the aVariable to cache table. |aVariable| must be a pointer for a
-  // static variable. The value will be modified when the pref value is changed
-  // but note that even if you modified it, the value isn't assigned to the
-  // pref.
-  static void AddBoolVarCache(bool* aVariable, const nsACString& aPref,
-                              bool aDefault = false);
-  template <MemoryOrdering Order>
-  static void AddAtomicBoolVarCache(Atomic<bool, Order>* aVariable,
-                                    const nsACString& aPref,
-                                    bool aDefault = false);
-  static void AddIntVarCache(int32_t* aVariable, const nsACString& aPref,
-                             int32_t aDefault = 0);
-  template <MemoryOrdering Order>
-  static void AddAtomicIntVarCache(Atomic<int32_t, Order>* aVariable,
-                                   const nsACString& aPref,
-                                   int32_t aDefault = 0);
-  static void AddUintVarCache(uint32_t* aVariable, const nsACString& aPref,
-                              uint32_t aDefault = 0);
-  template <MemoryOrdering Order>
-  static void AddAtomicUintVarCache(Atomic<uint32_t, Order>* aVariable,
-                                    const nsACString& aPref,
-                                    uint32_t aDefault = 0);
-  static void AddFloatVarCache(float* aVariable, const nsACString& aPref,
-                               float aDefault = 0.0f);
-
-  static void AddAtomicFloatVarCache(std::atomic<float>* aVariable,
-                                     const nsACString& aPref,
-                                     float aDefault = 0.0f);
-
-  template <int N>
-  static void AddBoolVarCache(bool* aVariable, const char (&aPref)[N],
-                              bool aDefault = false) {
-    return AddBoolVarCache(aVariable, nsLiteralCString(aPref), aDefault);
-  }
-  template <MemoryOrdering Order, int N>
-  static void AddAtomicBoolVarCache(Atomic<bool, Order>* aVariable,
-                                    const char (&aPref)[N],
-                                    bool aDefault = false) {
-    return AddAtomicBoolVarCache<Order>(aVariable, nsLiteralCString(aPref),
-                                        aDefault);
-  }
-  template <int N>
-  static void AddIntVarCache(int32_t* aVariable, const char (&aPref)[N],
-                             int32_t aDefault = 0) {
-    return AddIntVarCache(aVariable, nsLiteralCString(aPref), aDefault);
-  }
-  template <MemoryOrdering Order, int N>
-  static void AddAtomicIntVarCache(Atomic<int32_t, Order>* aVariable,
-                                   const char (&aPref)[N],
-                                   int32_t aDefault = 0) {
-    return AddAtomicIntVarCache<Order>(aVariable, nsLiteralCString(aPref),
-                                       aDefault);
-  }
-  template <int N>
-  static void AddUintVarCache(uint32_t* aVariable, const char (&aPref)[N],
-                              uint32_t aDefault = 0) {
-    return AddUintVarCache(aVariable, nsLiteralCString(aPref), aDefault);
-  }
-  template <MemoryOrdering Order, int N>
-  static void AddAtomicUintVarCache(Atomic<uint32_t, Order>* aVariable,
-                                    const char (&aPref)[N],
-                                    uint32_t aDefault = 0) {
-    return AddAtomicUintVarCache<Order>(aVariable, nsLiteralCString(aPref),
-                                        aDefault);
-  }
-  template <int N>
-  static void AddFloatVarCache(float* aVariable, const char (&aPref)[N],
-                               float aDefault = 0.0f) {
-    return AddFloatVarCache(aVariable, nsLiteralCString(aPref), aDefault);
-  }
-
-  template <int N>
-  static void AddAtomicFloatVarCache(std::atomic<float>* aVariable,
-                                     const char (&aPref)[N],
-                                     float aDefault = 0.0f) {
-    return AddAtomicFloatVarCache(aVariable, nsLiteralCString(aPref), aDefault);
-  }
-
   // When a content process is created these methods are used to pass changed
   // prefs in bulk from the parent process, via shared memory.
   static void SerializePreferences(nsCString& aStr);